### ISSUE TITLE
Split out the different auth options

### ISSUE THREAD

**Issue by singpolyma** [aaronpk/IndieAuth.com#120]
Why not have a separate endpoint for each auth option (sms, pgp, relmeauth, etc) -- this way I could delegate to particular ones I want to use like I can for indiecert etc

**Comment by aaronpk** [comment]
Hm, would you want to delegate to the `pgp` one to prevent any other login mechanisms from being used? One of the nice things about indieauth.com showing multiple options is that depending on the device you're logging in on, you might want to choose a different option. For example I usually use GitHub or GPG login when I'm on my main computer, but use Twitter from my phone.

I can definitely see value in wanting to limit the options provided by indieauth.com to a subset of the rel-me links on your site. (Maybe I want Twitter listed on my site, but never want to use it for login.) 

What about using the query string to indicate the supported providers?

`https://indieauth.com/auth?providers=github.com,pgp,sms` etc. In that case, indieauth.com could even present them to you in the order given. 

Similar to https://github.com/aaronpk/IndieAuth.com/issues/112, if only one is set then it could redirect immediately instead of making you click the button, which would be a better user experience.

**Comment by aaronpk** [comment]
Hm, would you want to delegate to the `pgp` one to prevent any other login mechanisms from being used? One of the nice things about indieauth.com showing multiple options is that depending on the device you're logging in on, you might want to choose a different option. For example I usually use GitHub or GPG login when I'm on my main computer, but use Twitter from my phone.

I can definitely see value in wanting to limit the options provided by indieauth.com to a subset of the rel-me links on your site. (Maybe I want Twitter listed on my site, but never want to use it for login.) 

What about using the query string to indicate the supported providers?

`https://indieauth.com/auth?providers=github.com,pgp,sms` etc. In that case, indieauth.com could even present them to you in the order given. 

Similar to https://github.com/aaronpk/IndieAuth.com/issues/112, if only one is set then it could redirect immediately instead of making you click the button, which would be a better user experience.

**Comment by singpolyma** [comment]
I like the menu for the reasons you say, but would like to limit what options are allowed. However I'd like to accept options not part of indieauth.com (such as indiecert). Why not make everything first-class (all distributed-indieauth providers) and make the main indieauth provider just a provider of a menu based on what options your domain advertises (with sane defaults if nothing is advertised, basically fall back to current UX)

**Comment by aaronpk** [comment]
Not that it's your fault, but I think you're starting to confuse the two roles of indieauth.com.

Role 1) indieauth.com is a service that developers can use to handle all the hard work of doing rel-me-auth with specific providers directly. In this case, the application developer has a trust relationship with indieauth.com and users should not be concerned that they're using indieauth.com, from their POV they are just signing in to the website. This is how the indiewebcamp.com wiki uses indieauth.com

Role 2) indieauth.com is a service that users can delegate their domain to. To use indieauth.com this way, the user links to indieauth.com as their `authorization_endpoint` on their domain. In this case, the user has a trust relationship with indieauth.com, and an application discovers the user's auth endpoint by following the rel link on their website. Micropub apps like Quill work this way, where you will only ever see indieauth.com if you have delegated to it yourself.

Does this help clear things up? In situation 2, you'll only ever see indieauth.com if you explicitly set it as your authorization endpoint. You could use indiecert.net or use your own auth server instead. In situation 1, where a developer has chosen to use indieauth.com instead of implementing authentication themselves, you're limited to the options that indieauth.com has implemented. However the idea is that indieauth.com implements a good number of options and in a secure way, making it a better option for developers than implementing PGP/SMS/GitHub/etc themselves.

With that in mind, could you rephrase your request in that context?

**Comment by singpolyma** [comment]
Right, so I'm specifically talking about option 2. Basically, it would be nice if I weren't tied down to either indieauth.com's options or another provider, but could pick-and-choose. That is, use SMS/PGP/whatever from indieauth.com but also indiecert, etc

I get a UX similar to this now on sites that use option 1 by using indiecert as my delegated provider, but I see the full range of indieauth.com options of course instead of a restricted set.

The idea would be to have a system where I use option2 to delegate to indieauth.com just as I can now, but I can specify (either with additional markup or some other way) which auth options (both internal ones to indieauth and external ones like indiecert) I want to see. My proposal is based on the idea that one way to achieve that would be to make the "internal" options first class wrt the "external" options and then just add a way to specify which ones to show (possibly by just having delegate links to all of them in my markup, with the one to indieauth.com first, followed by indieauth sms, followed by indiecert, etc)

Does the idea of what I'm hoping to eventually acheive at least make sense to you?

**Comment by aaronpk** [comment]
I've done something like this on the first replacement service: indielogin.com

https://indielogin.com/setup#choosing-auth-providers

That allows the website owner to mark only certain providers to be used rather than the full list of rel=me links. I'm going to close this issue though because this project is going to eventually be replaced by something that works very differently.
